### PR TITLE
Update `StringExt` implementation to use `str`

### PR DESCRIPTION
**Pull request by ever0de**
context: https://github.com/gluesql/gluesql/pull/999/files#diff-f4f27532f21aa04e9550528e5d3922ece5b9ac8d3056096eacfd11c7da2a7ed0R344-R345

### CODE REVIEW

File: core/src/data/literal.rs [hunk 1]
```diff
@@ -62,24 +62,48 @@ impl<'a> TryFrom<&'a AstLiteral> for Literal<'a> {
...
+    fn eq(&self, other: &&str) -> bool {
+        PartialEq::<str>::eq(self, *other)
+    }
+}
+
+impl PartialOrd<&str> for Literal<'_> {
+    fn partial_cmp(&self, other: &&str) -> Option<Ordering> {
+        PartialOrd::<str>::partial_cmp(self, *other)
+    }
+}
```
Comment (panarch) [inline core/src/data/literal.rs:105]: Can we use `AsRef<str>` here?
```rust
impl<T: AsRef<str>> PartialEq<T> for Literal<'_> {
     fn eq(&self, other: &T) -> bool {             
         PartialEq::<str>::eq(self, other.as_ref())
     }                                             
 }                                                 
```

Comment (ever0de) [reply core/src/data/literal.rs:105]: Oh.. `&&str` doesn't seem to be included here.

Comment (ever0de) [reply core/src/data/literal.rs:105]: No, I was impatient. (just before check success)

**Review by panarch — APPROVED** [review]
nice improvement! Thanks a lot 👍

**Pull request merged**